### DISCUSSION TITLE
focus-window capability in window-rule

### DISCUSSION THREAD

**Discussion by permezel** [YaLTeR/niri#731] [Ideas]
I would like to be able to focus the newly created window.
This would require support for:
   focus-workspace "desired-workspace"
or
  open-on-workspace --focus "desired-workspace"

within a window-rule.

It would also be nice to be able to control where in the workspace one ends up with the new window: left-most, right-most, or default.

My rational for this is that currently I have "terminal", "browse" and "emacs" named workspaces. Mod+Return spawns a terminal, but I then have to go looking for it, if I am currently in some other workspace.

I also sometimes want to override the workspace rule, and have the terminal be created in the current workspace.
If something like the floating window from `sway` were available, I would not want the override, as I would just push a terminal into the floating state, and use that.

**Comment by YaLTeR** [comment]
> Mod+Return spawns a terminal, but I then have to go looking for it, if I am currently in some other workspace.

This is the same issue as #558 and will be fixed there

> I also sometimes want to override the workspace rule, and have the terminal be created in the current workspace.

I believe you should be able to clear open-on-workspace in a later window-rule with `open-on-workspace null`

**Reply by permezel** [reply]
> > Mod+Return spawns a terminal, but I then have to go looking for it, if I am currently in some other workspace.
> 
> This is the same issue as #558 and will be fixed there

Per docs: "If the window opens on an output that is not currently focused, the window will not be automatically focused."
I read that to imply it would not switch the workspace.  My mistake.
> 
> > I also sometimes want to override the workspace rule, and have the terminal be created in the current workspace.
> 
> I believe you should be able to clear open-on-workspace in a later window-rule with `open-on-workspace null`

Not really what I was after, but I think binding another key seq to `"alacritty" "--class" "something-else"` will do what I want.

**Reply by YaLTeR** [reply]
> Per docs: "If the window opens on an output that is not currently focused, the window will not be automatically focused."
I read that to imply it would not switch the workspace. My mistake.

This does not apply to windows that explicitly get passed focus. Fixing #558 involves making niri explicitly pass focus to clients spawned with compositor binds.
